addons/website_sale/models/account_move.py

```diff
@@ -18,3 +18,14 @@ def preview_invoice(self):
             # URL should always be relative, safety check
             action['url'] = f'/@{action["url"]}'
         return action
+
+    @api.depends('date')
+    def _compute_website_id(self):
+        for move in self:
+            source_websites = move.line_ids.sale_line_ids.order_id.website_id
+            if len(source_websites) == 1:
+                move.website_id = source_websites
+            elif len(source_websites) > 1 and all(website.id == source_websites[0].id for website in source_websites):
+                move.website_id = source_websites[0]
```

Comment:
```suggestion
            if len(source_websites) == 1:
                move.website_id = source_websites
```
The orm will already aggregate identical websites, so if `len(source_websites > 1)` then there will be different websites in `source_websites`